libs/utils/wa_results_collector.py

```diff
@@ -206,6 +206,37 @@ def _read_wa_dir(self, wa_dir):
 
         # results.csv contains all the metrics reported by WA for all jobs.
         df = pd.read_csv(os.path.join(wa_dir, 'results.csv'))
+        # When using Monsoon, the device is a single channel which reports
+        # two metrics. This means that devlib's DerivedEnergymeasurements class
+        # cannot see the output. Due to the way that the monsoon.py script
+        # works, it looks difficult to change Monsoon over to the Acme way of
+        # operating. As a workaround, let's mangle the results here instead.
+        __metrics = df['metric'].unique()
```

Comment:
__metrics will be mangled into _WaResultsCollector__metrics by the Python parser since it appears inside a class definition block. That is not really harmful as this name is not really exposed, but that can still be confusing when looking at backtraces. The leading double underscore convention in the kernel does not mix well with Python unfortunately:
```
>>> class A:
...   def f(self):
...    print(__metric)
...
>>> a=A()
>>> a.f()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 3, in f
NameError: name '_A__metric' is not defined
```

Reply:
I changed the name to unique_metrics instead in this new version.